### ISSUE TITLE
Questiones about `decoder`

### ISSUE THREAD

**Issue by FantasyJXF** [ZQPei/DSSD#3]
The `forward` pass in `DSSD/dssd/modeling/decoder/decoder.py`, the `features[-2-level]` updates in each circle, did it do better than use the `ORIGINAL` feature comes from ResNet.

Another question, is this repo modified from mask-rcnn? The code architecture looks so familiar.

I'm really appreciate your work. Thank you very much.

**Comment by FantasyJXF** [comment]
Is there any pretrained model to test?

How's your evaluate results?

**Comment by ZQPei** [comment]
Thanks for telling me.
Indeed, it is modified from `mask-rcnn`. The pretrained model is not released because I think it is not too hard to train. The mAP result on VOC is around 0.8 .

**Comment by ZQPei** [comment]
Hold on a second, I don't think there's anything wrong with my code.
The `features[-2-level]` updates as `x_conv` for each loop and feed as `x_deconv` in the next one.
This is exactly that is done in DSSD.

**Comment by FantasyJXF** [comment]
<img width="670" alt="屏幕快照 2020-04-08 上午10 19 21" src="https://user-images.githubusercontent.com/16235526/78737588-78f12380-7982-11ea-9934-dbdd442152f0.png">

I didn't read original DSSD code, the picture on the paper seems didn't update `x_conv` during `deconvolution module`. 
By the way, I think this won't make much difference.

I port your code to `ssd.prtorch`, wait to see the results.

**Comment by ZQPei** [comment]
It would save memory.
